pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -67,6 +75,12 @@ func (endpointSliceStrategy) PrepareForUpdate(ctx context.Context, obj, old runt
 
 	newEPS.ObjectMeta = ogNewMeta
 	oldEPS.ObjectMeta = ogOldMeta
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceTerminatingCondition) {
+		for i := range newEPS.Endpoints {
+			newEPS.Endpoints[i].Conditions.Terminating = nil
```

Comment:
I think we need to maintain an existing value here if it is already set to ensure compatibility with future versions that could set this field.

Reply:
Those future versions would have the feature gate enabled so we wouldn't hit this code path right?

Reply:
I think the idea here is that we need to be able to safely round trip between API versions. As an example, if a resource was created with a 1.21 API Server, data from that resource should not be dropped/unset by a 1.20 API Server. At least that's what I had to do for [AppProtocol in 1.18](https://github.com/kubernetes/kubernetes/blob/release-1.18/pkg/apis/core/validation/validation.go#L4273-L4283) when it was alpha.

Reply:
Okay, this makes sense now, thanks for explaining! :)

Reply:
PR updated, based on your feedback